pkg/dtls/internal/udp/conn.go

```diff
@@ -105,7 +104,7 @@ readLoop:
 			n = copy(cBuf, buf[:n])
 			conn.sizeCh <- n
 		case <-conn.doneCh:
-			continue readLoop
+			return
```

Comment:
I get leaked routines with this, before it would deadlock on the ReadFrom above.

Not sure this is the right change, but wanted to throw it up here.

Reply:
This doesn't seem like the right approach. The readLoop should keep passing packets to the Conns until the last one is closed. We'll have to look into it further.